core/src/main/java/hudson/security/GlobalSecurityConfigurationInclude.java

```diff
@@ -0,0 +1,129 @@
+package hudson.security;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+
+import org.acegisecurity.context.SecurityContext;
+import org.acegisecurity.context.SecurityContextHolder;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import hudson.BulkChange;
+import hudson.Extension;
+import hudson.model.Describable;
+import hudson.model.Descriptor;
+import hudson.model.Descriptor.FormException;
+import hudson.security.AuthorizationStrategy.Unsecured;
+import hudson.security.csrf.DefaultCrumbIssuer;
+import hudson.util.FormApply;
+import jenkins.install.SetupWizard;
+import jenkins.install.SetupWizardSecurityConfiguration;
+import jenkins.install.TransientJenkins;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+
+/**
+ * A version of the GlobalSecurityConfiguration which can be included in another page 
+ * and configures a transient Jenkins as opposed to the global Jenkins instance.
+ * This is used by the {@link SetupWizard}
+ */
+public class GlobalSecurityConfigurationInclude implements Describable<GlobalSecurityConfigurationInclude> {
```

Comment:
Seems like the name of this class should be tied to `SetupWizard` since it seems quite specific to it.